From 8fc79f25a6e647ac038d6ff4c2fa9839623337a9 Mon Sep 17 00:00:00 2001
From: Jian-Hong Pan <jhp@endlessos.org>
Date: Tue, 15 Apr 2025 14:17:54 +0800
Subject: [PATCH 1/3] [Endless] linux-pam: Allow weak passwords

Allow weak passwords by dropping pam_pwquality module check and
use_authtok from pam_unix module. Besides, swapping pam_unix's
nullok_secure to nullok fixes both "sudo" and "polkit" prompts.

https://phabricator.endlessm.com/T14095
https://phabricator.endlessm.com/T35802
---
 files/linux-pam-config/password-auth | 3 +--
 files/linux-pam-config/system-auth   | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/files/linux-pam-config/password-auth b/files/linux-pam-config/password-auth
index a2803b492..1df8aac70 100644
--- a/files/linux-pam-config/password-auth
+++ b/files/linux-pam-config/password-auth
@@ -7,8 +7,7 @@ auth        required      pam_deny.so
 account     required      pam_unix.so
 
 -password   sufficient    pam_systemd_home.so
-password    requisite     pam_pwquality.so try_first_pass local_users_only retry=3 authtok_type=
-password    sufficient    pam_unix.so try_first_pass use_authtok nullok_secure sha512 shadow
+password    sufficient    pam_unix.so try_first_pass nullok sha512 shadow
 password    required      pam_deny.so
 
 -session    optional      pam_systemd_home.so
diff --git a/files/linux-pam-config/system-auth b/files/linux-pam-config/system-auth
index a2803b492..1df8aac70 100644
--- a/files/linux-pam-config/system-auth
+++ b/files/linux-pam-config/system-auth
@@ -7,8 +7,7 @@ auth        required      pam_deny.so
 account     required      pam_unix.so
 
 -password   sufficient    pam_systemd_home.so
-password    requisite     pam_pwquality.so try_first_pass local_users_only retry=3 authtok_type=
-password    sufficient    pam_unix.so try_first_pass use_authtok nullok_secure sha512 shadow
+password    sufficient    pam_unix.so try_first_pass nullok sha512 shadow
 password    required      pam_deny.so
 
 -session    optional      pam_systemd_home.so
-- 
2.49.0

